lib/rubocop/cop/ezcater/rspec_require_gql_error_helpers.rb

```diff
@@ -0,0 +1,35 @@
+module RuboCop
+  module Cop
+    module Ezcater
+      # Enforce use of GQLErrors helpers instead of throwing
+      # GraphQL::ExecutionErrors directly
+      #
+      # @example
+      #
+      #   # good
+      #   GQLErrors.summary_error("An error occurred")
+      #   GQLErrors.request_error("You can't access this", 401)
+      #   GQLErrors.field_error("is invalid", :first_name, "First Name")
+      #   GQLErrors.field_errors_for(my_model, context)
+      #   GQLErrors.field_errors_for(my_model, context, summary_error: "An error occurred")
+      #   GQLErrors.field_errors_for(my_model, context, field_mapping: { first: :first_name })
+      #
+      #   # bad
+      #   GraphQL::ExecutionError.new("An error occurred"
+      #   GraphQL::ExecutionError.new("You can't access this", options: {status_code: 401})
+      class RspecRequireGqlErrorHelpers < Cop
+        MSG = "Use the helpers provided by `GQLErrors` instead of raising `GraphQL::ExecutionError` directly.".freeze
+
+        def_node_matcher :graphql_const?, <<~PATTERN
+          (const (const _ :GraphQL) :ExecutionError)
```

Comment:
Should this check for calling `.new` on this class or should any reference to the class be forbidden?

For example, this cop would flag `rescue GraphQL::ExecutionError`.

Reply:
I think for now, I would like to start that way actually. I don't see a reason for us to be interacting with that class at all. I'd be curious on @christoomey's thoughts on this, but for now, I'd think I'd like to opt into leaving it. We can always loosen that restriction once we've come up with a legit use case for referencing the class. I can't think of any off the top of my head.

Reply:
@mjfaga I'm fine with taking that approach for now. Will you be disabling this cop for the file where you define the `GQLErrors` helpers?

If we're taking this approach, could you add a test example where you're just referencing the class (without `.new`) to document that you're intentionally flagging all references to it not just initializing new instances?

Reply:
Correct, this cop will be disabled in that 1 file.